ts/sagas/wallet.ts

```diff
@@ -343,6 +345,62 @@ function* showWalletOrSelectPsp(idWallet: number, paymentId?: string) {
   }
 }
 
+const MAX_RETRIES_POLLING = 20;
+const DELAY_BETWEEN_RETRIES_MS = 1000;
+
+const delay = (ms: number): Promise<undefined> =>
+  new Promise(resolve => setTimeout(resolve, ms));
+
+const pollForPaymentId = async (
+  backendClient: ReturnType<typeof BackendClient>,
+  p: { paymentContextCode: CodiceContestoPagamento },
+  retries = MAX_RETRIES_POLLING
+): Promise<string | undefined> => {
+  if (retries === 0) {
+    return undefined;
+  }
+  const response = await backendClient.getPaymentId(p);
+  if (response === undefined) {
+    return undefined;
+  }
+  if (response.status === 200) {
+    return response.value.idPagamento;
+  }
+  if (response.status === 404) {
+    await delay(DELAY_BETWEEN_RETRIES_MS);
+    console.warn("waiting");
+    return pollForPaymentId(backendClient, p, retries - 1);
+  }
+  return undefined;
+};
+
+// WIP: is this the appropriate place for this function?
+const attivaRpt = async (
+  sessionToken: SessionToken | undefined,
+  rptId: RptId,
+  paymentContextCode: CodiceContestoPagamento,
+  amount: AmountInEuroCents
+) => {
+  if (sessionToken) {
```

Comment:
what's the point of wrapping the whole function body with a check? this is a code smell, indicating this check should be the responsibility of the caller